mindmeld/components/entity_resolver.py

```diff
@@ -930,8 +1114,52 @@ def _construct_whitelist_query(entity, weight=1, use_phons=False):
 
             return results
 
-    def _load(self):
-        """Loads the trained entity resolution model from disk."""
+    def dump(self, path, **_kwargs):
+        """
+        Args:
+            path (str): A .pkl file path where the resolver will be dumped. The model hash will be
+                dumped at {path}.hash file path.
+        """
+        os.makedirs(os.path.dirname(path), exist_ok=True)
+        head, ext = os.path.splitext(path)
+        resolver_config_path = head + ".config" + ext
+        with open(resolver_config_path, "wb") as fp:
+            pickle.dump(self._resolver_configs, fp)
+            fp.close()
+
+        hash_path = path + ".hash"
+        os.makedirs(os.path.dirname(hash_path), exist_ok=True)
+        with open(hash_path, "w") as hash_file:
+            hash_file.write(self.hash)
+
+        self.dirty = False
+
+    def load(self, path=None):
+        """
+        Args:
+            path (str): A .pkl file path where the resolver has been dumped, if dumped by
+                calling .dump()
+        """
+        # Before adding the dump+load functionality, other entity resolvers' `load` method simply
+        # was redirecting to the `fit` method whereas for ES resolver, it is doing something
+        # different- hence the load method is overwritten for the ES resolver
+
+        # backwards compatability:
+        # previous versions of the code might try to access the `load` method upon creating indices
+        # w/o dumping the config or w/o passing the `path` argument
+        try:
+            head, ext = os.path.splitext(path)
+            resolver_config_path = head + ".config" + ext
+            with open(resolver_config_path, "rb") as fp:
+                _resolver_configs = pickle.load(fp)
+                fp.close()
+            self._use_double_metaphone = _resolver_configs["use_double_metaphone"]
+        except (TypeError, FileNotFoundError):
+            msg = f"Unable to load previously used configs for the {self.__class__.__name__} " \
+                  f"resolver while loading."
+            logger.warning(msg)
+
+        # elasticsearch loading
```

Comment:
i think we might need to have additional check on top of index existence check (maybe based on `dirty` flag and also the `hash`?) because we could have an existing index created based on different config than what we load from serialized file?

Also it looks like we are not loading the `hash` from dump()?

Reply:
Because we dump the processed entity map in the dump method and load it back in load method, we need not check for the hash equivalence as it will be the same data. This check is similar to other NLP components like domain classifier, etc.